snmp/datadog_checks/snmp/snmp.py

```diff
@@ -98,13 +106,49 @@ def _get_instance_key(self, instance):
 
         return key
 
+    def discover_instances(self):
+        network = ipaddress.ip_network(self._config.network_address)
+        discovery_interval = self._config.instance.get('discovery_interval', 3600)
+        discovery_pause = self._config.instance.get('discovery_pause', 1)
+        while self._running:
+            start_time = time.time()
+            for host in network.hosts():
+                host = str(host)
+                if host in self._config.discovered_instances:
+                    continue
+                instance = self._config.instance.copy()
+                instance.pop('network_address')
+                instance['ip_address'] = host
+
+                config = InstanceConfig(
+                    instance,
+                    self.warning,
+                    self.init_config.get('global_metrics', []),
+                    self.mibs_path,
+                    self.profiles,
+                    self.profiles_by_oid,
+                )
+                try:
+                    sys_object_oid = self.fetch_sysobject_oid(config)
+                except Exception:
+                    continue
+                if sys_object_oid not in self.profiles_by_oid:
+                    self.log.warn("Host %s didn't match a profile for sysObjectID %s", host, sys_object_oid)
```

Comment:
1) If the `Exception` above occur, it seems `sys_object_oid` would be unreferenced.

2) IMO we should log the error message above or append it to the `log.warn` below. Will help investigation and debugging.

Reply:
1. There is a continue, so we move on to the next host.

Reply:
1. Yes, you are right, I missed the continue :)